onnxruntime/core/providers/cuda/reduction/reduction_ops.cc

```diff
@@ -127,6 +338,7 @@ static Status PrepareForReduce(OpKernelContext* ctx,
   }
 
   Tensor* Y = ctx->Output(0, TensorShape(squeezed_output_dims));
+  CUDA_RETURN_IF_ERROR(cudaMemset(Y->MutableDataRaw(), 0, Y->SizeInBytes()));
   *y_pp = Y;
```

Comment:
Why is zeroing out the memory required?

Reply:
This reduction keep adding values to this buffer.

Reply:
Can we capture this info in a comment so it's clear to the next developer why the cudaMemset is needed?

Reply:
will add it